### PR TITLE
feat: human feedback form

### PR DESCRIPTION
**Pull request by datnguyennnx**
- Create feedback table. 
- Add seed file for insert feedback type. 
- Define enum type for FE.
- Feedback form for like and dislike rating. 
- Api createRating using integrationProcedure. That coverage case is an unknown user or user through apiToken. 
- Create dashboard page for chat history.
- **You should migrate and seed the database.**

image
image
image

### CODE REVIEW

File: src/server/api/routers/feedback.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { z } from 'zod'
+import * as schema from '~/server/db/migration/schema'
+
+import { createTRPCRouter, integrationProcedure } from '~/server/api/trpc'
+import { db } from '~/server/db'
+import { FeedbackTypeEnum } from '~/model/feedback'
+import { uuidv7 } from 'uuidv7'
+
+export const feedbackRouter = createTRPCRouter({
+  createRating: integrationProcedure
```
Comment (taynguyen) [inline src/server/api/routers/feedback.ts:10]: Should be `createFeedback` to consistent the naming.

File: src/server/api/routers/feedback.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    .input(
+      z.object({
+        chatId: z.string(),
+        feedbackType: z.nativeEnum(FeedbackTypeEnum),
+        notes: z.string(),
```
Comment (taynguyen) [inline src/server/api/routers/feedback.ts:15]: notes could be optional.

File: src/server/db/migration/schema.ts [hunk 3]
```diff
@@ -379,6 +379,37 @@ export const chats = createTable(
...
+      .references(() => chats.id),
+    typeId: integer('feedback_type').notNull(),
+    notes: text('notes'),
+    createdAt: timestamp('created_at').notNull().defaultNow(),
+    createdBy: uuid('created_by').references(() => users.id),
+    updatedAt: timestamp('updated_at'),
+    updatedBy: uuid('updated_by').references(() => users.id),
+  },
+  (chat_feedback) => ({
+    botIdIdx: index('chat_feedback_id_idx').on(chat_feedback.chatId),
```
Comment (taynguyen) [inline src/server/db/migration/schema.ts:397]: May need to fix the naming here. 
Could close other PR. Seem this PR already cover last PR.

**Review by taynguyen — COMMENTED** [review]
minor comment

File: src/server/api/routers/chat.ts [hunk 4]
```diff
@@ -43,7 +43,7 @@ function getListHandler() {
 
       const chats = await db.query.chats.findMany({
         where: eq(schema.chats.threadId, input.threadId),
-        orderBy: [desc(schema.chats.id)],
```
Comment (taynguyen) [inline src/server/api/routers/chat.ts:46]: Why we need to change loading chat from oldest to newest ?

File: src/server/db/migration/schema.ts [hunk 5]
```diff
@@ -379,6 +380,44 @@ export const chats = createTable(
...
+  {
+    id: uuid('id').notNull().primaryKey(),
+    threadId: uuid('thread_id')
+      .notNull()
+      .references(() => threads.id),
+    chatId: uuid('chat_id')
+      .notNull()
+      .references(() => chats.id),
+    typeId: integer('feedback_type').notNull(),
+    isLike: boolean('isLike'),
```
Comment (taynguyen) [inline src/server/db/migration/schema.ts:394]: Why we need a new column for this ?
We could add type_id: good feed back.

**Review by taynguyen — COMMENTED** [review]
Some comments

File: src/server/api/routers/chat.ts (hunk 4 above)
Comment (datnguyennnx) [reply src/server/api/routers/chat.ts:46]: We make confuse at here, table will render newest to oldest thread chat. => Then i replace "asc" to "desc" at api/routers/thread.ts files.  
DetailDrawer will show oldest to newest chat like chat history by using "asc" at api/routers/chat.ts. 
image

File: src/server/api/routers/feedback.ts [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+          threadId: input.threadId,
+          chatId: input.chatId,
+          typeId: input.feedbackType,
+          notes: input.notes,
+          createdAt: new Date(),
+        })
+        .returning()
+    }),
+
+  getList: integrationProcedure
```
Comment (taynguyen) [inline src/server/api/routers/feedback.ts:35]: We shouldn't allow integrated users to get list of feedbacks
Should be admin only. Let's use protectedProcedure

File: src/server/api/routers/feedback.ts [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+        .select({
+          count: sql<number>`COUNT(*)`,
+        })
+        .from(schema.chat_feedback)
+        .where(eq(schema.chat_feedback.threadId, input.threadId))
+      const count = Number(countRow[0]?.count) ?? 0
+
+      const chat_feedback = await db.query.chat_feedback.findMany({
+        where: eq(schema.chat_feedback.threadId, input.threadId),
+        orderBy: [asc(schema.chat_feedback.id)],
```
Comment (taynguyen) [inline src/server/api/routers/feedback.ts:54]: This is getting the oldest feedbacks first. 
Could consider to get latest first or not ?
Or the FE need sort options, we could consider support sending order from FE as well.

**Review by taynguyen — COMMENTED** [review]
Minor comments.

**Review by taynguyen — APPROVED** [review]

**Pull request merged**